src/app/services/cpe-api.service.ts

```diff
@@ -0,0 +1,109 @@
+import {Injectable} from "@angular/core";
+import {HttpClient} from "@angular/common/http";
+import {Observable} from "rxjs";
+
+const projectsGetRoute = "http://localhost:3000/projects/get/"
+const projectsRoute = "http://localhost:3000/projects"
+const pagesGetRoute = "http://localhost:3000/pages/get/"
+const pagesRoute = "http://localhost:3000/pages"
+const widgetTypesGetRoute = "http://localhost:3000/widgettypes/get/"
+const widgetTypeRoute = "http://localhost:3000/widgettype"
+
+/**
+ * The base interface for all resources.
+ */
+export interface iCpeResource {
+    id: string;
+    label: string;
+    description: string;
+}
+
+/**
+ * The interface for getting projects from the api service
+ */
+export interface iProject extends iCpeResource {
+    mainPageID: string;
+    pages: string[];
+}
+
+/**
+ * The interface for getting projects from the api service
+ */
+export interface iPage extends iCpeResource {
+    gridDimension: Dimension
+    header: Widget[];
+    body: Widget[];
+    footer: Widget[]
+}
+
+type Widget = {
```

Comment:
Why is it a type here? I would prefer to have it streamlined with iProject asf. as an interface like so:

export interface iPage extends iCpeResource {
    gridDimension: Dimension
    header: Widget[];
    body: Widget[];
    footer: Widget[];
}

Reply:
Good thougth: https://github.com/dasch-swiss/dsp-cpe/pull/15/commits/00a40d88d5d09301501a7806d30725e5ea76d96b